### PR TITLE
Clarify requirements for add-on version list

### PR DESCRIPTION
**Pull request by tofumatt**
I'm bad at Sphinx docs so apologies if this is the wrong format, but I was reading
the API docs and found this section a bit hard to parse. There are a lot of commas,
"and"s, and "or"s. I feel like this makes it a bit easier to grok.

### CODE REVIEW

File: docs/topics/api/addons.rst [hunk 1]
```diff
@@ -119,9 +119,9 @@ Detail
 This endpoint allows you to fetch a specific add-on by id, slug or guid.
 
     .. note::
-        Non-public add-ons, or add-ons with only unlisted versions, require
-        authentication and either reviewer permissions or a user account listed
-        as a developer of the add-on.
+        Non-public add-ons and add-ons with only unlisted versions require both:
+            * authentication
```
Comment (eviljeff) [inline docs/topics/api/addons.rst]: Have you compiled the sphinx docs to check it likes `*` as a bullet point? 'tis a fickle beast.

Comment (eviljeff) [reply docs/topics/api/addons.rst]: Github's auto-preview indicates it displays the bullet points but makes the preceding sentence bold, which looks weird.  Github doesn't know about our extensions and home brewed styles either.

Comment (tofumatt) [reply docs/topics/api/addons.rst]: screenshot 2017-09-10 17 41 37

It looks like this, so seemingly good.

**Comment by tofumatt** [comment]
Cool, makes sense. I updated that note everywhere in this file.

File: docs/topics/api/addons.rst (hunk 1 above)
Comment (tofumatt) [reply docs/topics/api/addons.rst]: Oh, I tweaked it so it's not bold.
